src/Sulu/Component/Rest/ListBuilder/Doctrine/DoctrineListBuilder.php

```diff
@@ -597,4 +586,109 @@ public function createOrExpression(array $expressions)
 
         throw new InvalidExpressionArgumentException('or', 'expressions');
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function addSelectField(FieldDescriptorInterface $fieldDescriptor)
+    {
+        if (!$fieldDescriptor instanceof DoctrineFieldDescriptorInterface) {
+            return;
+        }
+
+        return parent::addSelectField($fieldDescriptor);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function addField(FieldDescriptorInterface $fieldDescriptor)
+    {
+        if (!$fieldDescriptor instanceof DoctrineFieldDescriptorInterface) {
+            return;
+        }
+
+        return parent::addField($fieldDescriptor);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function addSearchField(FieldDescriptorInterface $fieldDescriptor)
+    {
+        if (!$fieldDescriptor instanceof DoctrineFieldDescriptorInterface) {
+            return;
+        }
+
+        return parent::addSearchField($fieldDescriptor);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function addFieldDescriptor(FieldDescriptorInterface $fieldDescriptor)
+    {
+        if (!$fieldDescriptor instanceof DoctrineFieldDescriptorInterface) {
+            return;
+        }
+
+        parent::addFieldDescriptor($fieldDescriptor);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function addGroupBy(FieldDescriptorInterface $fieldDescriptor)
+    {
+        if (!$fieldDescriptor instanceof DoctrineFieldDescriptorInterface) {
+            return;
+        }
+
+        parent::addGroupBy($fieldDescriptor);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setSelectFields($fieldDescriptors)
+    {
+        parent::setSelectFields(
+            array_filter(
+                $fieldDescriptors,
+                function (FieldDescriptorInterface $field) {
+                    return $field instanceof DoctrineFieldDescriptorInterface;
+                }
+            )
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setFields($fieldDescriptors)
+    {
+        parent::setFields(
+            array_filter(
+                $fieldDescriptors,
+                function (FieldDescriptorInterface $field) {
+                    return $field instanceof DoctrineFieldDescriptorInterface;
+                }
+            )
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setFieldDescriptors(array $fieldDescriptors)
+    {
+        parent::setFieldDescriptors(
+            array_filter(
+                $fieldDescriptors,
+                function (FieldDescriptorInterface $field) {
+                    return $field instanceof DoctrineFieldDescriptorInterface;
+                }
+            )
+        );
+    }
```

Comment:
These additions are also a bug hiding procedure for me... Was there a specific reason to add this? If we really have to introduce that, I would prefer throwing exceptions, telling the developer that something wrong has been passed.